homework/index.js

```diff
@@ -30,17 +32,152 @@
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+  function addRowToTable(parent, rowElements = []) {
+    const row = createAndAppend('tr', parent);
+    rowElements.forEach(elem => {
+      if (elem.type === 'th') {
+        createAndAppend('th', row, { text: elem.text });
+      } else if (elem.type === 'tableDataWithLink') {
+        const tableDataElement = createAndAppend('td', row);
+        createAndAppend('a', tableDataElement, {
+          text: elem.text,
+          href: elem.href,
+          target: '_blank',
+        });
+      } else if (elem.type === 'td') {
+        createAndAppend('td', row, { text: elem.text });
       }
     });
   }
 
+  function buildRepositoryInfo(repository, containerDiv) {
+    const table = createAndAppend('table', containerDiv, {
+      id: 'repository-table',
+    });
+    const tableBody = createAndAppend('tbody', table);
+    addRowToTable(tableBody, [
+      { type: 'th', text: 'Repository:' },
+      {
+        type: 'tableDataWithLink',
+        text: repository.name,
+        href: repository.html_url,
+        target: '_blank',
+      },
+    ]);
+    addRowToTable(tableBody, [
+      { type: 'th', text: 'Description:' },
+      {
+        type: 'td',
+        text: repository.description,
+      },
+    ]);
+    addRowToTable(tableBody, [
+      { type: 'th', text: 'Forks:' },
+      {
+        type: 'td',
+        text: repository.forks_count,
+      },
+    ]);
+    addRowToTable(tableBody, [
+      { type: 'th', text: 'Last updated:' },
+      {
+        type: 'td',
+        text: new Date(repository.updated_at).toLocaleDateString(),
+      },
+    ]);
+  }
+
+  function buildRepositoryContributors(contributors, containerDiv) {
+    createAndAppend('h2', containerDiv, { text: 'Contributions' });
+    const contributorsList = createAndAppend('ul', containerDiv, { id: 'contributors-list' });
+    contributors.forEach(contributor => {
+      const listItem = createAndAppend('li', contributorsList, {
+        // class: 'contributor-list-item',
+      });
+      const container = createAndAppend('div', listItem, {
+        class: 'contributor-container',
+      });
+      createAndAppend('img', container, {
+        src: contributor.avatar_url,
+        alt: 'contributor avatar',
+        class: 'contributor-avatar',
+      });
+      const infoContainer = createAndAppend('div', container, {
+        class: 'contributor-info-container',
+      });
+      const nameContainer = createAndAppend('div', infoContainer);
+      const contributorName = createAndAppend('h3', nameContainer);
+      createAndAppend('a', contributorName, {
+        text: contributor.login,
+        href: contributor.html_url,
+        target: '_blank',
+      });
+      const numberOfContributionsContainer = createAndAppend('div', infoContainer);
```

Comment:
You can probably get rid of the `nameContainer` and `numberOfContributionsContainer` divs:

```html
<div>
  <h3>
    <a href="https://github.com/wouterkleijn" target="_blank">wouterkleijn</a>
  </h3>
</div>
<div>
  <h3>8</h3>
</div>
```